lib/client/ca_export_test.go

```diff
@@ -528,3 +501,97 @@ func (m *multiCAAuthClient) PerformMFACeremony(
 	// Skip MFA ceremonies.
 	return nil, &mfa.ErrMFANotRequired
 }
+
+func TestExportIntegrationAuthorities(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	testAuth, err := auth.NewTestAuthServer(auth.TestAuthServerConfig{
+		ClusterName: "localcluster",
+		Dir:         t.TempDir(),
+	})
+	require.NoError(t, err)
+
+	fingerprint, err := sshutils.AuthorizedKeyFingerprint([]byte(fixtures.SSHCAPublicKey))
+	require.NoError(t, err)
+
+	mockedAuthClient := &mockAuthClient{
+		server: testAuth.AuthServer,
+		integrationsClient: mockIntegrationsClient{
+			caKeySet: &types.CAKeySet{
+				SSH: []*types.SSHKeyPair{{
+					PublicKey: []byte(fixtures.SSHCAPublicKey),
+				}},
+			},
+		},
+	}
+
+	for _, tc := range []struct {
+		name        string
+		req         ExportIntegrationAuthoritiesRequest
+		checkError  require.ErrorAssertionFunc
+		checkOutput func(*testing.T, []*ExportedAuthority)
+	}{
+		{
+			name: "missing integration",
+			req: ExportIntegrationAuthoritiesRequest{
+				Type: "github",
+			},
+			checkError: require.Error,
+		},
+		{
+			name: "unknown type",
+			req: ExportIntegrationAuthoritiesRequest{
+				Type:        "unknown",
+				Integration: "integration",
+			},
+			checkError: require.Error,
+		},
+		{
+			name: "github",
+			req: ExportIntegrationAuthoritiesRequest{
+				Type:        "github",
+				Integration: "integration",
+			},
+			checkError: require.NoError,
+			checkOutput: func(t *testing.T, authorities []*ExportedAuthority) {
+				require.Len(t, authorities, 1)
+				require.Contains(t, string(authorities[0].Data), fixtures.SSHCAPublicKey)
+			},
+		},
+		{
+			name: "matching fingerprint",
+			req: ExportIntegrationAuthoritiesRequest{
+				Type:             "github",
+				Integration:      "integration",
+				MatchFingerprint: fingerprint,
+			},
+			checkError: require.NoError,
+			checkOutput: func(t *testing.T, authorities []*ExportedAuthority) {
+				require.Len(t, authorities, 1)
+				require.Contains(t, string(authorities[0].Data), fixtures.SSHCAPublicKey)
+			},
+		},
+		{
+			name: "no matching fingerprint",
+			req: ExportIntegrationAuthoritiesRequest{
+				Type:             "github",
+				Integration:      "integration",
+				MatchFingerprint: "something-does-not-match",
+			},
+			checkError: require.NoError,
+			checkOutput: func(t *testing.T, authorities []*ExportedAuthority) {
+				require.Len(t, authorities, 1)
+				require.Empty(t, authorities[0].Data)
```

Comment:
This returns an authority with no data? Shouldn't this error / return zero authorities?

Reply:
Empty output is the current behavior of `ExportAuthorityFingerprint` for host/user CA. But when i write the UT, i feel an error is more appropriate as well. Updated.